Search Feature Implemented

yusuferdemnacar opened the pull request:
### Summary
The post search feature is implemented. The users can now search for posts by a list of keywords and/or a list of labels.

The endpoint requires no authentication.

### Detailed Content

* two endpoints for the functionality
  * contmgr/searchpost
  * contmgr/searchpost/
* Proper JSON responses with status codes for various cases
  * 200: Search successful
  * 400: Search failed due to no parameters.
* Unit tests using a temporary SQLite database
  * Tests for each scenario
* API documentation
  * Prepared with the same data used in the tests for conformality

### Tasks
- [x] Implementing search
- [x] Writing tests
- [x] Creating API documentation

### Related Issues

* #296

artun-akdogan commented:
I have reviewed the code and it looks OK. I will also run it on my local computer and test it. If no errors were found, I will approve this pull request.

yusuferdemnacar commented:
I updated the responses in accordance with the notes in #296, 7ec3075fd4ac887d2fd61342e07089512d98cd73.
The tests did not need an update.
I also updated the responses in the API document, 0e97152d63eaeed0f7da4c0554a9f5eba04b4f59.

artun-akdogan reviewed (approved):
Code works without an issue. Thanks for the work
